Portal/src/Datahub.Infrastructure/Services/UsersStatusService.cs

```diff
@@ -0,0 +1,46 @@
+﻿using Datahub.Application.Configuration;
+using Datahub.Application.Services;
+using Microsoft.Extensions.Logging;
+using Newtonsoft.Json;
+
+namespace Datahub.Infrastructure.Services
+{
+    public partial class UsersStatusService : IUsersStatusService
+    {
+        private readonly ILogger<UsersStatusService> _logger;
+        private readonly IHttpClientFactory _httpClientFactory;
+        private readonly DatahubPortalConfiguration _datahubPortalConfiguration;
+        
+        public UsersStatusService(ILogger<UsersStatusService> logger, IHttpClientFactory httpClientFactory,
+            DatahubPortalConfiguration datahubPortalConfiguration)
+        {
+            _logger = logger;
+            _httpClientFactory = httpClientFactory;
+            _datahubPortalConfiguration = datahubPortalConfiguration;
+        }
+        
+        public async Task<Dictionary<string, List<string>>> GetUsersStatus()
+        {
+            var url = _datahubPortalConfiguration.DatahubGraphUsersStatusFunctionUrl;
+            
+            var numberOfRetries = 0;
+            const int maxNumberOfRetries = 5;
+            string resultString;
+            
+            do
+            {
+                using var client = _httpClientFactory.CreateClient();
```

Comment:
I think you shouldn't dispose the httpClients (or check if this is what is recommended), I may be wrong but the httpClientFactory is who manages the lifecycle of the httpClients.

Reply:
In most recent commit, moved the httpclient outside of the while loop and removed `using` keyword to let the factory manage the lifecycle